src/jit/gentree.cpp

```diff
@@ -17906,26 +17907,9 @@ bool GenTreeHWIntrinsic::OperIsMemoryLoad()
             {
                 return false;
             }
-            else // We have 3 or more operands/args
+            else if (HWIntrinsicInfo::isAVX2GatherIntrinsic(gtHWIntrinsicId))
```

Comment:
Here could be simplified to 
```cpp

        else if (category == HW_Category_IMM)
        {
            return HWIntrinsicInfo::isAVX2GatherIntrinsic(gtHWIntrinsicId);
        }
```

Reply:
The whole thing looks overly complicated. I don't know why we need to check for "maybe memory load" and for categories instead of just checking for the few special cases - gathers and broadcasts.

Reply:
> The whole thing looks overly complicated.

I agree. I'm going to log a bug for this and see about following up in a future PR.

Reply:
Logged https://github.com/dotnet/coreclr/issues/22897